Sandbox failed (error starting container: exit status 125)

I have error with default installation

### Command

```scripts/run_analysis.sh -ecosystem pypi -package Django```

```root ➜ /workspaces/package-analysis (main) $ scripts/run_analysis.sh -ecosystem pypi -package Django
-----------------------------------------
Package Details
Ecosystem:                pypi
Package:                  Django
Version:                  
Location:                 remote
-----------------------------------------
Analysing package

2023-03-13T10:57:57.754Z        INFO    worker/logging.go:63    Got request     {"ecosystem": "pypi", "name": "Django", "version": "", "package_path": "", "results_bucket_override": ""}
2023-03-13T10:57:57.786Z        INFO    analyze/main.go:199     Starting static analysis
2023-03-13T10:57:57.786Z        DEBUG   sandbox/init.go:71      Creating bridge network
2023-03-13T10:57:57.794Z        DEBUG   sandbox/init.go:31      Loading iptable rules
2023-03-13T10:57:57.797Z        INFO    worker/runstatic.go:35  Running static analysis {"tasks": ["all"]}
2023-03-13T10:57:57.797Z        DEBUG   sandbox/sandbox.go:240  podman  {"args": ["--cgroup-manager=cgroupfs", "--events-backend=file", "pull", "gcr.io/ossf-malware-analysis/static-analysis:latest"]}
2023-03-13T10:57:58.730Z        DEBUG   sandbox/sandbox.go:240  podman  {"args": ["--cgroup-manager=cgroupfs", "--events-backend=file", "image", "prune", "-f"]}
2023-03-13T10:57:58.860Z        DEBUG   sandbox/sandbox.go:240  podman  {"args": ["--cgroup-manager=cgroupfs", "--events-backend=file", "create", "--runtime=/usr/local/bin/runsc_compat.sh", "--init", "--dns=8.8.8.8", "--dns=8.8.4.4", "--dns-search=.", "--network=analysis-net", "-v", "/results.json:/results.json", "gcr.io/ossf-malware-analysis/static-analysis:latest"]}
2023-03-13T10:57:59.194Z        DEBUG   sandbox/sandbox.go:240  podman  {"args": ["--cgroup-manager=cgroupfs", "--events-backend=file", "start", "--runtime=/usr/local/bin/runsc_compat.sh", "--runtime-flag=root=/var/run/runsc", "--runtime-flag=debug-log=/tmp/sandbox_logs_3054061372/runsc.log.%COMMAND%", "2eba113554b38dbf232e29d446f3e7c18cfd2831ca721daf37e20143649610d3"]}
2023-03-13T10:57:59.258Z        WARN    log/writer.go:63        time="2023-03-13T10:57:59Z" level=warning msg="Couldn't run auplink before unmount /var/lib/containers/storage/aufs/mnt/d1d267d24ebbf46d29baa0a5a8eec260674db6b43604655a7595aa5d63e5391b: exec: \"auplink\": executable file not found in $PATH"    {"args": ["/usr/local/bin/staticanalyze", "-ecosystem", "pypi", "-package", "django", "-version", "4.1.7", "-analyses", "all", "-output", "/results.json"]}
github.com/ossf/package-analysis/internal/log.WriteTo
        /src/internal/log/writer.go:63
github.com/ossf/package-analysis/internal/log.Writer.func1
        /src/internal/log/writer.go:40
2023-03-13T10:57:59.449Z        WARN    log/writer.go:63        Error: unable to start container "2eba113554b38dbf232e29d446f3e7c18cfd2831ca721daf37e20143649610d3": error mounting storage for container 2eba113554b38dbf232e29d446f3e7c18cfd2831ca721daf37e20143649610d3: error creating aufs mount to /var/lib/containers/storage/aufs/mnt/d1d267d24ebbf46d29baa0a5a8eec260674db6b43604655a7595aa5d63e5391b: invalid argument    {"args": ["/usr/local/bin/staticanalyze", "-ecosystem", "pypi", "-package", "django", "-version", "4.1.7", "-analyses", "all", "-output", "/results.json"]}
github.com/ossf/package-analysis/internal/log.WriteTo
        /src/internal/log/writer.go:63
github.com/ossf/package-analysis/internal/log.Writer.func1
        /src/internal/log/writer.go:40
2023-03-13T10:57:59.451Z        DEBUG   sandbox/sandbox.go:240  podman  {"args": ["--cgroup-manager=cgroupfs", "--events-backend=file", "stop", "-t=5", "-i", "2eba113554b38dbf232e29d446f3e7c18cfd2831ca721daf37e20143649610d3"]}
2023-03-13T10:57:59.581Z        DEBUG   sandbox/sandbox.go:240  podman  {"args": ["--cgroup-manager=cgroupfs", "--events-backend=file", "rm", "--all", "--force"]}
2023-03-13T10:57:59.914Z        FATAL   analyze/main.go:128     Static analysis aborted {"error": "sandbox failed (error starting container: exit status 125)"}
main.staticAnalysis
        /src/cmd/analyze/main.go:128
main.main
        /src/cmd/analyze/main.go:200
runtime.main
        /usr/local/go/src/runtime/proc.go:250

-----------------------------------------
Analysis failed

docker process exited with code 1

Ecosystem:                pypi
Package:                  Django
Version:                  
Location:                 remote
-----------------------------------------```

--- a/scripts/run_analysis.sh
+++ b/scripts/run_analysis.sh
@@ -115,7 +115,39 @@ fi
 
 DOCKER_OPTS=("run" "--cgroupns=host" "--privileged" "--rm")
 
-DOCKER_MOUNTS=("-v" "/var/lib/containers:/var/lib/containers" "-v" "$RESULTS_DIR:/results" "-v" "$STATIC_RESULTS_DIR:/staticResults" "-v" "$FILE_WRITE_RESULTS_DIR:/writeResults" "-v" "$LOGS_DIR:/tmp")
+# On development systems, we mount /var/lib/containers so that sandbox images can be
+# shared between the host system and the analysis image. However, this requires the
+# directory to be backed by a non-overlay filesystem.
+# In some environments, e.g. GitHub Codespaces, this is not the case, and we need to
+# specify a different mount dir which is backed by a non-overlay filesystem.
+
+function is_overlay_mount() {
+	if [[ $(findmnt -T "$1" -n -o FSTYPE) == "overlay" ]]; then
+		echo "true"
+	else
+		echo "false"
+	fi
+}
+
+CONTAINER_MOUNT_DIR="/var/lib/containers"
+
+if [[ -n "$CONTAINER_DIR_OVERRIDE" ]]; then
+	CONTAINER_MOUNT_DIR="$CONTAINER_DIR_OVERRIDE"
+elif [[ $CODESPACES == "true" ]]; then
+	CONTAINER_MOUNT_DIR=$(mktemp -d)
+	echo "GitHub Codespaces environment detected, using $CONTAINER_MOUNT_DIR for container mount"
+elif [[ $(is_overlay_mount /var/lib) == true ]]; then
+	if [[ $(is_overlay_mount /tmp) == false ]]; then
+		CONTAINER_MOUNT_DIR=$(mktemp -d)
+		echo "Warning: /var/lib is an overlay mount using $CONTAINER_MOUNT_DIR for container mount"
+	else
+		echo "Environment error: /var/lib is an overlay mount, please set CONTAINER_DIR_OVERRIDE to a directory that is backed by a non-overlay filesystem"
+		exit 1
+	fi
+fi
+
+
+DOCKER_MOUNTS=("-v" "$CONTAINER_MOUNT_DIR:/var/lib/containers" "-v" "$RESULTS_DIR:/results" "-v" "$STATIC_RESULTS_DIR:/staticResults" "-v" "$FILE_WRITE_RESULTS_DIR:/writeResults" "-v" "$LOGS_DIR:/tmp")
 
 ANALYSIS_IMAGE=gcr.io/ossf-malware-analysis/analysis
 

--- a/scripts/run_analysis.sh
+++ b/scripts/run_analysis.sh
@@ -121,11 +121,12 @@ DOCKER_OPTS=("run" "--cgroupns=host" "--privileged" "--rm")
 # In some environments, e.g. GitHub Codespaces, this is not the case, and we need to
 # specify a different mount dir which is backed by a non-overlay filesystem.
 
-function is_overlay_mount() {
-	if [[ $(findmnt -T "$1" -n -o FSTYPE) == "overlay" ]]; then
-		echo "true"
+# Checks that the given mountpoint has the given filesystem mount type
+function is_mount_type() {
+	if [[ $(findmnt -T "$2" -n -o FSTYPE) == "$1" ]]; then
+		return 0
 	else
-		echo "false"
+		return 1
 	fi
 }
 
@@ -136,10 +137,10 @@ if [[ -n "$CONTAINER_DIR_OVERRIDE" ]]; then
 elif [[ $CODESPACES == "true" ]]; then
 	CONTAINER_MOUNT_DIR=$(mktemp -d)
 	echo "GitHub Codespaces environment detected, using $CONTAINER_MOUNT_DIR for container mount"
-elif [[ $(is_overlay_mount /var/lib) == true ]]; then
-	if [[ $(is_overlay_mount /tmp) == false ]]; then
+elif is_mount_type overlay /var/lib; then
+	if is_mount_type overlay /tmp && ! is_mount_type tmpfs /tmp; then
 		CONTAINER_MOUNT_DIR=$(mktemp -d)
-		echo "Warning: /var/lib is an overlay mount using $CONTAINER_MOUNT_DIR for container mount"
+		echo "Warning: /var/lib is an overlay mount, using $CONTAINER_MOUNT_DIR for container mount"
 	else
 		echo "Environment error: /var/lib is an overlay mount, please set CONTAINER_DIR_OVERRIDE to a directory that is backed by a non-overlay filesystem"
 		exit 1

